dataclients/kubernetes/clusterstate_test.go

```diff
@@ -0,0 +1,60 @@
+package kubernetes
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/zalando/skipper/dataclients/kubernetes/definitions"
+)
+
+var letterRunes = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+
+func randStringRunes(n int) string {
+	b := make([]rune, n)
+	for i := range b {
+		b[i] = letterRunes[rand.Intn(len(letterRunes))]
+	}
+	return string(b)
+}
+
+func benchmarkCachedEndpoints(b *testing.B, n int) {
+	endpoints := make(map[definitions.ResourceID]*endpoint)
+	for i := 0; i < n; i++ {
+		name := randStringRunes(10)
+		if i == 0 {
+			name = "postgres"
+		}
+		rid := newResourceID("default", name)
+		ep := &endpoint{
+			Meta:    nil,
+			Subsets: nil,
+		}
+		endpoints[rid] = ep
+	}
+
+	cs := &clusterState{
+		ingresses:   nil,
+		ingressesV1: nil,
+		routeGroups: nil,
+		services:    nil,
+		endpoints:   endpoints,
+		secrets:     nil,
+	}
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		cs.GetEndpointsByTarget("default", "postgres", "TCP", &definitions.BackendPort{})
```

Comment:
It could be a bit tricky to write proper benchmark due to compile optimizations, see
* https://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go
* https://github.com/golang/go/issues/27400
* https://github.com/golang/go/issues/48768

Lets assign result to the global "sink" variable as suggested here https://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go#compiler-optimisation